app/javascript/components/clients/Doctorappointment.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable no-alert */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import './Bookappointment.css';
+
+const DoctorAppointment = () => {
+  const { id } = useParams();
+  const appointments = useSelector((state) => state.appointmentReducer);
+
+  const onSubmit = (e) => {
+    document.querySelector('.popup-box').style.display = 'none';
+    const date = document.getElementById('appointment_date').value;
```

Comment:
As we're using React components, I am probably gonna request you to instead of using direct references to the DOM (i.e, using document.getElementById) that you figure out how you can use React (w/c uses the virtual DOM) to perform object references and manipulation.